### ISSUE TITLE
vm docker create must take .pub files

### ISSUE THREAD

**Issue by squillace** [Azure/azure-xplat-cli#2393]
We need to treat `azure vm create` scenarios not only securely but identically. This means that with linux VM creations, we need to be able to generate/pass .pub files to each one. 

1. azure vm quick-create (https://github.com/Azure/azure-xplat-cli/issues/2135)
2. azure vm create (https://github.com/Azure/azure-xplat-cli/issues/1710)
3. azure vm docker create

This issue is to fix number 3; moving forward, we need to treat "creation" as a single thing, and not each creation command as a different thing, despite differences. The following text is yanked from 1710:

@ruslangabitov @ahmetalpbalkan I note in https://github.com/Azure/azure-xplat-cli/issues/1710#issuecomment-123204193 that we are STILL specifying .pem files in testing (bad; they include private information). Here's the help for `azure vm create` in arm right now:

```
virtual-machines [sshtypo] % azure vm create --help
help:    Create a VM
help:
help:    Usage: vm create [options] <dns-name> <image> [userName] [password]
<snip />
help:      -t, --ssh-cert <openssh-rsa-file|pem-file|fingerprint>  the SSH certificate
```

No one should use .pem files as certificate files because they contain private information. We have heard this, and we now take ssh rsa format .pub files! Yay! BUT: here's the `azure vm docker create` command. Why are we still taking pem files for the ssh-key in  `docker create vm` command? 

```
azure vm docker create --help
<snip />
help:      -t, --ssh-cert <pem-file|fingerprint>  the SSH certificate
```

can we have unity? parallelism? can we update all creation commands at one and the same time? Should I file a separate issue there?

**Comment by ahmetalpbalkan** [comment]
cc: @sauryadas would love to hear your input as well

**Comment by ahmetalpbalkan** [comment]
@squillace I think `vm docker create` is just a special case of `vm create`. I'd say let's change title of this issue to drop `docker` keyword. Because #1710 is entirely about CLI itself creating the ssh keys and managing them for you (you don't even see a ssh word while creating the VM). I'd like to keep that issue like that.

**Comment by squillace** [comment]
@ahmetalpbalkan I agree that it's a special case. But as it turns out, quick-create is also a special case, and that was treated with a separate bug. We added ssh-rsa support to arm vm create, and then did NOT add it to docker create, and nothing, not one thing in #1710 mentions paying any attention to docker create. In short, the myopia is strong, Luke.

Now, if you request it and think it's appropriate to close this and modify #1710, I'm good with that. Go ahead and ACK and I'll close and add to #1710.

**Comment by ahmetalpbalkan** [comment]
I was saying this issue should stay open without `docker` keyword. Maybe we can merge `vm quick-create` and `vm create`, but #1710 has nothing to do with accepting .pub files as inputs. It's about getting rid of .pub files entirely.

**Comment by squillace** [comment]
Ah, I misunderstood. I'm with you. Modifying.

**Comment by squillace** [comment]
@ahmetalpbalkan sound about right? Or should I cut down to clarify further?

**Comment by ahmetalpbalkan** [comment]
@squillace this is getting super confusing. I thought you were simply asking for --ssh-cert argument accepting id_rsa.pub file, now you're talking about generation of the file (overlaps with #1710) and "handle by default if exist" (I'm guessing you're saying use existing one if one present at $HOME/.ssh, which is not a good idea). 

Feel free to call me sometime. :)

**Comment by squillace** [comment]
Right. emailed you. My intention by referencing #1710 was to say that whatever implementation was decided there, it should also apply to quick-create and docker create. Having individual issues for what is a general "azure vm create" problem with all three commands is as confusing to me as anything else. :-)
